libraries/chain/db_management.cpp

```diff
@@ -220,7 +220,8 @@ void database::close(bool rewind)
    // DB state (issue #336).
    clear_pending();
 
-   object_database::flush();
+   if (is_clean)
+      object_database::flush();
```

Comment:
I think it's better to change the parameter name to `do_flush` or similar.

Reply:
agree with you `do_flush` name is better and makes more sense